### PR TITLE
RecoveryServices.Backup SDK for backup management service version 2016-12-01

### PR DESCRIPTION
**Pull request by DheerendraRathor**
## Description
This PR contains newly generated code based on new [swagger spec](https://github.com/Azure/azure-rest-api-specs/blob/master/arm-recoveryservicesbackup/2016-12-01/swagger/backupManagement.json) with api-version 2016-12-01

Other Changes:
- Updated TestSettings to use new Vm, Resource Group and Storage
- Removed redundant code from TestSettings.cs
- Fixed test case disposal in ItemScenarioTests.cs
- Fixed enable protection test case in TestHelper.cs

new swagger spec PR: https://github.com/Azure/azure-rest-api-specs/pull/1139 

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as a link to the swagger spec, used to generate the code.
- [x] The `project.json` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by shahabhijeet** [comment]
@azuresdkci add to whitelist

File: src/SDKs/RecoveryServices.Backup/Management.RecoveryServices.Backup/project.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "version": "1.2.0-preview",
+  "version": "1.3.0-preview",
```
Comment (shahabhijeet) [inline src/SDKs/RecoveryServices.Backup/Management.RecoveryServices.Backup/project.json]: @DheerendraRathor all project.json, global.json and *.xproj files should be deleted. We no longer need them.

**Review by shahabhijeet — DISMISSED** [review]
I do not see any changes to the csproj for the sdk. You have breaking change, so bumping off the number is necessary.
If you are in preview, a minor version bump is needed, but if you are in non-preview, then a major version bump will be required.

**Comment by DheerendraRathor** [comment]
Hi @shahabhijeet can you please review it?

**Comment by shahabhijeet** [comment]
@azuresdkci add to white list

**Comment by shahabhijeet** [comment]
@azuresdkci  test this please

**Comment by shawns1** [comment]
@azuresdkci please test

**Comment by PhaniKTalluri** [comment]
Signed off

**Comment by DheerendraRathor** [comment]
@shahabhijeet @shawns1 Any update on this?

**Comment by markcowl** [comment]
@azuresdkci add to whitelist

File: src/SDKs/RecoveryServices.Backup/Management.RecoveryServices.Backup/Generated/BackupEnginesOperationsExtensions.cs [hunk 2]
```diff
@@ -1,15 +1,18 @@
-// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (markcowl) [inline src/SDKs/RecoveryServices.Backup/Management.RecoveryServices.Backup/Generated/BackupEnginesOperationsExtensions.cs]: The copyright header has disappeared - this is *required*

File: src/SDKs/RecoveryServices.Backup/Management.RecoveryServices.Backup/Generated/BackupJobsOperationsExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,111 @@
+// Code generated by Microsoft (R) AutoRest Code Generator 1.0.1.0
```
Comment (markcowl) [inline src/SDKs/RecoveryServices.Backup/Management.RecoveryServices.Backup/Generated/BackupJobsOperationsExtensions.cs]: Copyright header

**Review by markcowl — DISMISSED** [review]
Need the licesne header in the generated files.  This should be the defult, but you cna update the behavior by adding 

--license-header=MICROSOFT_MIT on the command line, or in the config file [license-header: MICROSOFT_MIT](the config file would be a better place)

**Comment by shahabhijeet** [comment]
@azuresdkci test this please

**Pull request merged**